src/driver/driver.cpp

```diff
@@ -171,22 +173,42 @@ void EvolutionDriver::OutputCycleDiagnostics() {
   const int dt_precision = std::numeric_limits<Real>::max_digits10 - 1;
   if (tm.ncycle_out != 0) {
     if (tm.ncycle % tm.ncycle_out == 0) {
-      if (Globals::my_rank == 0) {
-        std::uint64_t zonecycles =
-            (pmesh->mbcnt - mbcnt_prev) *
-            static_cast<std::uint64_t>(pmesh->GetNumberOfMeshBlockCells());
-        std::cout << "cycle=" << tm.ncycle << std::scientific
-                  << std::setprecision(dt_precision) << " time=" << tm.time
-                  << " dt=" << tm.dt << std::setprecision(2) << " zone-cycles/wsec = "
-                  << static_cast<double>(zonecycles) / timer_cycle.seconds();
-        // insert more diagnostics here
-        std::cout << std::endl;
-
-        // reset cycle related counters
-        timer_cycle.reset();
-        // need to cache number of MeshBlocks as AMR/load balance change it
-        mbcnt_prev = pmesh->mbcnt;
+      std::uint64_t zonecycles =
+          (pmesh->mbcnt - mbcnt_prev) *
+          static_cast<std::uint64_t>(pmesh->GetNumberOfMeshBlockCells());
+      const auto time_cycle_all = timer_cycle.seconds();
+      const auto time_cycle_step = time_cycle_all - time_LBandAMR;
+      std::cout << "cycle=" << tm.ncycle << std::scientific
+                << std::setprecision(dt_precision) << " time=" << tm.time
+                << " dt=" << tm.dt << std::setprecision(2) << " zone-cycles/wsec_step="
+                << static_cast<double>(zonecycles) / time_cycle_step
+                << " wsec_step=" << time_cycle_step;
```

Comment:
Don't you need the `if (Globals::my_rank == 0)` flag still? I understand we might want block information per rank, but it's going to produce an unreadable mass of output when running at scale. We should at least be able to control that somehow.

Reply:
I double checked an this function is always only called for `Globals::my_rank == 0` which is why the original `if` had no meaning.
I have no strong opinion on where this happens (at the calling site or in this function) but I think that it should only be checked once.

Reply:
Ah ok. So long as it's called only for `Globals::my_rank==0` I'm happy.